### DISCUSSION TITLE
Can we add same commandLine " options name" for Base and Subcommand

### DISCUSSION THREAD

**Discussion by navinko** [remkop/picocli#1923] [Q&A] [answered]
I am getting below issue once i run my test cases.

picocli.CommandLine$DuplicateOptionAnnotationsException: Option name '--id' is used by both field String com.dps.core.audit.cli.ReportSubcommand.uuid and field String com.dps.core.audit.cli.Pos.sourceId

	at picocli.CommandLine$DuplicateOptionAnnotationsException.create(CommandLine.java:17963)
	at picocli.CommandLine$DuplicateOptionAnnotationsException.access$2500(CommandLine.java:17957)

Is there any workaround exist?

**Comment by remkop** [comment, marked as answer]
In a single command, it's not possible to have multiple options with the same name. 

When you say Base and subCommand, do you mean that the `subCommand` class extends the `Base` class, so the `subCommand` command inherits all options declared in the `Base` class?

You need to be a bit careful in that scenario to avoid using option names in the subclass that are already used in the superclass. You'll end up with a subclass that has two separate options with the same name and picocli doesn't allow that. 

The only solution is to use unique option names.

**Reply by navinko** [reply]
Thanks @remkop for quick response!
